.github/workflows/nightly_e2e.yml

```diff
@@ -0,0 +1,70 @@
+name: nightly_e2e
+
+on:
+  workflow_dispatch:
+    inputs:
+      flintlock_version:
+        type: "string"
+        required: false
+      capmvm_version:
+        type: "string"
+        required: false
+      capmvm_repo:
+        type: "string"
+        required: false
+      capmvm_branch:
+        type: "string"
+        required: false
+  schedule:
+    - cron: 0 0 * * *
+
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    env:
+      PROJECT_NAME: "liquid-metal-acceptance-tests"
+    name: e2e tests
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: hashicorp/setup-terraform@v2
+      with:
+        terraform_version: 1.2.5
+
+    - name: Run tests
+      env:
+        PROJECT_NAME: ${{ env.PROJECT_NAME }}
+        METAL_AUTH_TOKEN: ${{ secrets.METAL_AUTH_TOKEN }}
+        METAL_ORG_ID: ${{ secrets.EQUINIX_ORG_ID }}
+        FLINTLOCK_VERSION: ${{ inputs.flintlock_version }}
+      run: |
+        # make all E2E_ARGS="--version ${{ inputs.capmvm_version }} --repo ${{ inputs.capmvm_repo }} --branch ${{ inputs.capmvm_branch }}"
+        # This is hardcoded for the moment while I figure out if dispatch inputs only work on main
+        make all E2E_ARGS="--repo https://github.com/Callisto13/cluster-api-provider-microvm --branch e2e"
```

Comment:
When would this be removed?

Reply:
could be removed now, but i will probably wait until #1 is in first so i can test the inputs on main

Reply:
okay i can merge this now, but without seeing the "run workflow" before we get onto main, it may not work. but i can fix that after the fact i hope